scripts/check-tags.ts

```diff
@@ -9,27 +9,38 @@ interface Project {
   tags: string[]
 }
 
-const checkTags = async (filePath: string) => {
+const hasValidTags = async (filePath: string) => {
   const projectText = await readFile(filePath, 'utf8')
   const project = JSON.parse(projectText) as Project
   const tagsRaw = project.tags ?? []
   const tags = tagsRaw.map((tag) => tag.split(':'))
-  const isNormal =
-    (tags.length === 1 && tags[0][0] === 'scope') ||
-    (tags.length === 2 &&
-      tags.find(([key]) => key === 'scope') &&
-      tags.find(([key]) => key === 'lib') &&
-      tags[0][1] === tags[0][1])
+
   const isEmpty = tags.length === 0
+  // Only allow 'scope:*' or 'lib:*' tags
+  const validPrefix = tags.every(([key]) => key === 'scope' || key === 'lib')
+  // Must have at least one scope tag or be empty
+  const hasScopePrefix = tags.some(([key]) => key === 'scope') || isEmpty
+  // Are all tags the same?
+  const singularTag = new Set(tags.map(([, value]) => value)).size === 1
 
-  if (isEmpty) {
-    console.log(chalk.red.underline(filePath))
-    console.log('Missing nx tags for project boundaries\n')
+  // Exit early for good projects
+  if (!isEmpty && validPrefix && hasScopePrefix && singularTag) {
     return true
-  } else if (!isNormal) {
-    console.log(chalk.yellow.underline(filePath))
-    console.log('Unexpected nx tags:', tagsRaw.join(','), '\n')
   }
+  console.log(chalk.red.underline(filePath))
+  if (isEmpty) {
+    console.log('- Missing NX tags for project boundaries')
+  }
+  if (!hasScopePrefix) {
+    console.log('- Missing scope tag')
+  }
+  if (!validPrefix) {
+    console.log('- Unexpected NX tags')
+  }
+  if (!singularTag) {
+    console.log('- All tags must be the same')
+  }
```

Comment:
The strategy was to force devs to add tags (crash CI if empty) but only warn "not normal" projects.

We have a lot of "projects" which are "not normal" and this change will break CI for them. It is quite a lot of work to fix these abnormal projects, often requiring large-scale refactors, which the teams haven't had time for yet. Not sure how to prioritise that work...

Reply:
The list of misbehaving projects isn't too big. They're all only missing a `scope:*` tag. Seems like adding `react-next` or `react-spa` is the most permissive, which should be an "easy fix".

If it's not blocking in CI, when will it ever get fixed?

All misbehaving projects:
```
  'libs/university-gateway/project.json',
  'libs/service-portal/restrictions/project.json',
  'libs/clients/statistics/project.json',
  'libs/clients/signature-collection/project.json',
  'libs/clients/ship-registry/project.json',
  'libs/clients/sessions/project.json',
  'libs/clients/icelandic-government-institution-vacancies/project.json',
  'libs/clients/housing-benefit-calculator/project.json',
  'libs/clients/health-directorate/project.json',
  'libs/clients/auth/ids-api/project.json',
  'libs/clients/auth/admin-api/project.json',
  'libs/clients/aircraft-registry/project.json',
  'libs/api/domains/watson-assistant-chat/project.json',
  'libs/api/domains/auth-admin/project.json'
  ```

Reply:
Attempted to add the relevant scopes. `yarn check-tags` passes now locally. Let's see if he rest of CI passes as well :crossed_fingers: